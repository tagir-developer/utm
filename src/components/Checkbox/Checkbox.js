import React from 'react'

const Checkbox = ({ checkbox, checkedSource, index, device, changeCheckedSource }) => {

	// const inputRef = useRef(1)


	// useEffect(() => {
	// 	if (checkbox === checkedSource) {
	// 		inputRef.current.setAttribute('checked', '')
	// 	}
	// })

	const radioHandler = () => {
		changeCheckedSource(checkbox)
	}

	return (
		<div className="custom-control custom-radio custom-control-inline cursor-pointer">
			<input
				type="radio"
				// ref={inputRef}
				id={`customRadioInline-${device + index}`}
				name={`customRadioInline-${device}`}
				className="custom-control-input cursor-pointer"
				onChange={radioHandler}
				checked={checkbox === checkedSource ? true : false}

			/>
			<label className="custom-control-label cursor-pointer" htmlFor={`customRadioInline-${device + index}`}>
				{checkbox}
			</label>
		</div>
	)
}

Checkbox.defaultProps = {
	checkbox: '1', 
	checkedSource: '1', 
	index: 0, 
	device: 'device', 
	changeCheckedSource: () => {}
}

export default Checkbox