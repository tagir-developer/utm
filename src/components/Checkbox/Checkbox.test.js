import React from 'react'
import Checkbox from './Checkbox'

describe('<Checkbox />', () => {

	let wrapper

	beforeEach(() => {
		wrapper = shallow(<Checkbox />)
	})

	describe('Component render. Component props testing', () => {

		it('Should Checkbox component render with default props', () => {
			expect(toJson(wrapper)).toMatchSnapshot()
		})


		it('Should render custom checkbox name', () => {
			wrapper.setProps({
				checkbox: 'test-name'
			})
			expect(wrapper.find('label').text()).toBe('test-name')
		})

		it('Should set "checked={true}" attribute if "checkedSource === checkbox"', () => {
			wrapper.setProps({
				checkbox: 'checked-test',
				checkedSource: 'checked-test', 
			})
			expect(wrapper.find('input').props().checked).toBe(true)
		})

		it('Should set "checked={false}" attribute if "checkedSource !=== checkbox"', () => {
			wrapper.setProps({
				checkbox: 'checked-test',
				checkedSource: 'checked-another', 
			})
			expect(wrapper.find('input').props().checked).toBe(false)
		})

		it('Should render right attributes "htmlFor" and "id"', () => {
			wrapper.setProps({
				index: 7,
				device: 'test-device', 
			})
			expect(wrapper.find('input').props().id).toBe('customRadioInline-test-device7')
			expect(wrapper.find('label').props().htmlFor).toBe('customRadioInline-test-device7')
		})

	})

	describe('Checkbox component handlers testing', () => {

		it('Should call "changeCheckedSource"', () => {
			const changeHandler = jest.fn()
			wrapper.setProps({
				changeCheckedSource: changeHandler,
			})
			expect(changeHandler).toHaveBeenCalledTimes(0)
			wrapper.find('input').simulate('change')
			expect(changeHandler).toHaveBeenCalledTimes(1)
		})

		it('Should use default handler', () => {
			const result = Checkbox.defaultProps.changeCheckedSource()
			expect(result).toBe(undefined)
		})

	})


})