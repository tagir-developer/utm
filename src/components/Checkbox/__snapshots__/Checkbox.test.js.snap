// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> Component render. Component props testing Should Checkbox component render with default props 1`] = `
<div
  className="custom-control custom-radio custom-control-inline cursor-pointer"
>
  <input
    checked={true}
    className="custom-control-input cursor-pointer"
    id="customRadioInline-device0"
    name="customRadioInline-device"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="custom-control-label cursor-pointer"
    htmlFor="customRadioInline-device0"
  >
    1
  </label>
</div>
`;
